airflow/creds/aws_ssm.py

```diff
@@ -0,0 +1,127 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+"""
+Objects relating to sourcing connections from AWS SSM Parameter Store
+"""
+
+from typing import List
+
+import boto3
+
+from airflow import conf
+from airflow.creds import CONN_ENV_PREFIX, BaseCredsBackend
+from airflow.models import Connection
+
+
+class AwsSsmCredsBackend(BaseCredsBackend):
+    """
+    Retrieves Connection object from AWS SSM Parameter Store
+
+    Configurable via airflow config like so:
+
+    .. code-block:: ini
+
+        [aws_ssm_creds]
+        ssm_prefix = /airflow
+        profile_name = default
+
+    For example, if ssm path is ``/airflow/AIRFLOW_CONN_SMTP_DEFAULT``, this would be accessible if you
+    provide ``ssm_prefix = /airflow`` and conn_id ``smtp_default``.
+
+    """
+
+    conf_section = "aws_ssm_creds"
```

Comment:
```suggestion
    CONF_SECTION = "aws_ssm_creds"
```

etc to "mark" these as constants?

Reply:
updated